tasks/main.yml

```diff
@@ -35,6 +35,21 @@
     interval: 5
     token: "{{ nodeping_api_token }}"
     notifications: "{{ nodeping_notifications }}"
+  when: item.0.state is not defined or item.0.state == 'present'
   with_together:
     - "{{ nodeping_checks }}"
     - "{{ check_in_place.results }}"
+
+- name: Delete NodePing checks for target host
+  delegate_to: localhost
+  nodeping:
+    action: "delete"
+    checkid: "{{ item.1.ansible_facts.result | first }}"
+    token: "{{ nodeping_api_token }}"
+  when:
+    - item.1.ansible_facts.result | length > 0
+    - item.0.state is defined
+    - item.0.state != 'present'
+  with_together:
+    - "{{ nodeping_checks }}"
+    - "{{ check_in_place.results }}"
```

Comment:
Please add a newline at the end of the file.

Reply:
Sure I'll do that.